docs/how-xtrabackup-works.md

```diff
@@ -1,123 +1,133 @@
+advances the log sequence number (LSN). The server blocks the writes 
+during the process. The server determines when to purge the log based on the redo log consumption.
+
+
+Percona XtraBackup works by remembering the LSN when it starts and then 
+copying away the data files. It takes some time to do this, so if the files change, they reflect the 
+state of the database at different points. At the same time, Percona XtraBackup runs a background 
+process that watches the transaction log files and copies changes from them. 
+Percona XtraBackup does this continually because the
+transaction logs are written round-robin and can be reused. Percona XtraBackup needs the transaction log record for every change to the data files since it began execution.
+
+## Locks
+
+Locks are used in a backup to ensure data consistency. You cannot insert new data or modify data during the backup. Locking can cause performance issues, especially for frequently accessed tables.
+
+Percona XtraBackup uses `LOCK INSTANCE FOR BACKUP` to take a backup log for the entire process, including InnoDB and non-InnoDB data. The `xtrabackup` binary copies the InnoDB data and then copies the MyISAM tables and files. 
+
+The `xtrabackup` binary backs up the following file types:
+
+* .FRM
+
+* .MRG
+
+* .MYD
+
+* .MYI
+
+* .CSM
+
+* .CSV
+
+* .SDI
+
+* .PAR 
+
+### LOCK INSTANCE FOR BACKUP
+
+The `LOCK INSTANCE FOR BACKUP` statement is a feature introduced in MySQL 8.0 that allows you to acquire an instance-level backup lock. Executing the `LOCK INSTANCE FOR BACKUP` statement prevents any other session from acquiring a backup lock at the instance level. It ensures no changes to the data files while the backup process is running. This statement is executed after backing up all InnoDB/XtraDB data and logs and locks all non-InnoDB tables to safely copy non-InnoDB data without blocking any Data Manipulation Language (DML)  queries that modify InnoDB tables. The `LOCK INSTANCE FOR BACKUP` statement provides a more efficient and targeted way to lock the necessary tables during the backup process, reducing the impact on the database's availability and performance.
+
+The `BACKUP_ADMIN` privilege is required to execute the `LOCK INSTANCE FOR BACKUP` statement. This privilege manages database backups, ensures data consistency, and performs backup-related tasks.
+
+### Locking the binary log
+
+The `xtrabackup` binary uses `LOCK BINLOG FOR BACKUP` to block all operations that might change the binary log position or related information. The xtrabackup binary copies the redo log files, fetches the binary log coordinates, and `xtrabackup` unlocks the binary log and tables.
+
+The binary log position is printed to `STDERR`, and xtrabackup exits, returning 0 (zero) if successful. Be aware that the `STDERR` of xtrabackup does not write in any file. Redirect the error stream to a file to capture the binary log position. Use the following command, for example.
```

Comment:
Be aware that the `STDERR` of xtrabackup does not write in any file. - This sentence is confusing.

Reply:
Hmm, this is 5.7 way. In 8.0, we don't use this LOCK BINLOG FOR BACKUP. we use performance_schema.log_status and get the consistent binlog position across SEs, binlog, redolog.

So we have rewrite it to mention log_status.

Also, regarding the binlog pos, it is not necessary to parse from stderr/error log. xtrabackup stores this info in xtrabackup_binlog_info